src/mem/cache/tags/indexing_policies/IndexingPolicies.py

```diff
@@ -44,6 +44,9 @@ class BaseIndexingPolicy(SimObject):
     # Get the associativity
     assoc = Param.Int(Parent.assoc, "associativity")
 
+    # number of bits in the tag
+    tag_bits = Param.Unsigned(64, "number of bits in the tag")
```

Comment:
Updated. Please let me know what you think.

Reply:
Can this new parameter be added in another PR instead? This PR currently does:
- Creates a new non-secure cache (1)
- Updates the prefetchers to use the new cache (1)
- Updates the btbs to use the new cache (2)
- Adds a new parameter to indexing policy (3)

I have added numbers as a suggestion to split this PR up into multiple PRs. The idea is to make it easier to understand which functions were added to the cache because of the prefetcher caches, and which were added because of the BTB, so that we can try to come up with a good API.